apps/examples/testcase/le_tc/network/tc_net_poll.c

```diff
@@ -0,0 +1,244 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/*  @file tc_net_poll.c
+ * @brief Test Case Example for lwip_poll() API
+ */
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <poll.h>
+#include <pthread.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netutils/netlib.h>
+#include <lwip/sockets.h>
+#include <tinyara/os_api_test_drv.h>
+#include "tc_internal.h"
+
+struct poll_s {
+	int fd;
+	int sd;
+};
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll in multi-thread env
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_multi_thread_poll_p(int ret, int sd)
+{
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+	TC_SUCCESS_RESULT();
+}
+
+void *wait_event(void *arg)
+{
+	struct pollfd fds[1];
+	struct poll_test_args args;
+	struct poll_s *poll = (struct poll_s *)arg;
+	int sd = poll->sd;
+	int fd = poll->fd;
+	int ret = -1;
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	args.fd = sd;
+	args.fds = fds;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	tc_net_multi_thread_poll_p(ret, sd);
+
+	return NULL;
+}
+
+void *multi_thread_poll(void *arg)
+{
+	int sd = -1;
+	int fd = *(int *)arg;
+	pthread_t thread1;
+	struct poll_s args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		printf("socket fail %s:%d", __FUNCTION__, __LINE__);
+		return NULL;
+	}
+
+	args.fd = fd;
+	args.sd = sd;
+	pthread_create(&thread1, NULL, wait_event, (void *)&args);
```

Comment:
I think you need to check result of pthread_create.

Reply:
i don't think, there is any need to check result of pthread_create because if thread is not created so function will not be called, then socket will be closed and that TC will not execute.